manifold/src/boolean3.cpp

```diff
@@ -325,6 +325,7 @@ struct Kernel02 {
       if (forward) {
         if (!Shadows(vertPos.z, z02, expandP * vertNormalP[p0].z)) s02 = 0;
       } else {
+        ALWAYS_ASSERT(closestVert != -1, topologyErr, "No closest vert");
```

Comment:
So, the CI can't test CUDA since it doesn't have a GPU, so I tested it and it failed. I took a guess and commented out this line and it works. Personally, I think that's a failure of the CUDA compiler, since this line shouldn't do anything, but I think the problem stems from the fact that you can't throw an exception from inside a GPU kernel. Why it lets it compile, then totally breaks when run, I don't understand.

Reply:
You'll notice I have occasional `printf` lines in these where I ought to have asserts; that's why.

Reply:
Shall we use this instead? https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#assertion

Reply:
Yes, probably so. But let's save that for another PR since it's in a bunch of places. Also, I should probably make that change, since I can test it. I can't remember if there was a reason I'm not already using that.

Reply:
OK I will just comment this out for now then.

Reply:
I gave `assert` a try and now I remember where I got stuck. I can't seem to make Thrust catch the cuda error and throw an exception like I would expect. I can't find any documentation on it, so I wonder if it was never implemented?